client/modules/IDE/components/Editor.jsx

```diff
@@ -331,6 +351,100 @@ class Editor extends React.Component {
     this._cm.execCommand('findPersistent');
   }
 
+  showHint(_cm) {
+    if (!this.props.autocompleteHinter) {
+      CodeMirror.showHint(_cm, () => {});
+      return;
+    }
+
+    let focusedLinkElement = null;
+    const setFocusedLinkElement = (set) => {
+      if (set && !focusedLinkElement) {
+        const activeItemLink = document.querySelector(
+          `.CodeMirror-hint-active a`
+        );
+        if (activeItemLink) {
+          focusedLinkElement = activeItemLink;
+          focusedLinkElement.classList.add('focused-hint-link');
+          focusedLinkElement.parentElement.parentElement.classList.add(
+            'unfocused'
+          );
+        }
+      }
+    };
+    const removeFocusedLinkElement = () => {
+      if (focusedLinkElement) {
+        focusedLinkElement.classList.remove('focused-hint-link');
+        focusedLinkElement.parentElement.parentElement.classList.remove(
+          'unfocused'
+        );
+        focusedLinkElement = null;
+        return true;
+      }
+      return false;
+    };
+
+    const hintOptions = {
+      _fontSize: this.props.fontSize,
+      completeSingle: false,
+      extraKeys: {
+        'Shift-Right': (cm, e) => {
+          const activeItemLink = document.querySelector(
+            `.CodeMirror-hint-active a`
+          );
+          if (activeItemLink) activeItemLink.click();
+        },
+        Right: (cm, e) => {
+          setFocusedLinkElement(true);
+        },
+        Left: (cm, e) => {
+          removeFocusedLinkElement();
+        },
+        Up: (cm, e) => {
+          const onLink = removeFocusedLinkElement();
+          e.moveFocus(-1);
+          setFocusedLinkElement(onLink);
+        },
+        Down: (cm, e) => {
+          const onLink = removeFocusedLinkElement();
+          e.moveFocus(1);
+          setFocusedLinkElement(onLink);
+        },
+        Enter: (cm, e) => {
+          if (focusedLinkElement) focusedLinkElement.click();
+          else e.pick();
+        }
+      },
+      closeOnUnfocus: false
+      // completeOnSingleClick: false,
```

Comment:
It would be great to remove some unused, commented out lines like these!

Reply:
Removed!